clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -3,14 +3,28 @@ package io.treeverse.clients
 import com.google.common.cache.CacheBuilder
 import org.json4s._
 import org.json4s.native.JsonMethods._
-import scalaj.http.Http
+import io.treeverse.lakefs.clients.api
 
 import java.net.URI
 import java.time.Duration
 import java.util.concurrent.TimeUnit
 import java.util.concurrent.Callable
 
+private object ApiClient {
+  def translateS3(uri: URI): URI =
+    if (uri.getScheme == "s3") new URI("s3a", uri.getHost, uri.getPath, uri.getFragment) else uri
```

Comment:
worried that we do not include all the URI parts (ex: getPort getQuery) for cases where the endpoint is not amazon aws.

Reply:
Good idea.  Changed to use the MAX-ARGS [CONSTRUCTOR](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html#URI(java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20java.lang.String,%20java.lang.String,%20java.lang.String))